src/Servers/Kestrel/Core/src/Internal/Http2/FlowControl/FlowControl.cs

```diff
@@ -1,57 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-
-using System.Diagnostics;
-
-namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.FlowControl;
-
-/// <summary>
-/// Represents the flow control window for an <see cref="InputFlowControl"/>.
-/// Mutated as the available quota changes.
-/// </summary>
-/// <remarks>
-/// "FlowControlWindow" would probably be a clearer name.
-/// </remarks>
-internal struct FlowControl
-{
-    public FlowControl(uint initialWindowSize)
-    {
-        Debug.Assert(initialWindowSize <= Http2PeerSettings.MaxWindowSize, $"{nameof(initialWindowSize)} too large.");
-
-        Available = (int)initialWindowSize;
-        IsAborted = false;
-    }
-
-    public int Available { readonly get; private set; }
-    public bool IsAborted { readonly get; private set; }
-
-    public void Advance(int bytes)
-    {
-        Debug.Assert(!IsAborted, $"({nameof(Advance)} called after abort.");
-        Debug.Assert(bytes == 0 || (bytes > 0 && bytes <= Available), $"{nameof(Advance)}({bytes}) called with {Available} bytes available.");
-
-        Available -= bytes;
-    }
-
-    // bytes can be negative when SETTINGS_INITIAL_WINDOW_SIZE decreases mid-connection.
```

Comment:
If I'm reading this comment correctly, allowing it to go negative is important.  How can it be a `uint` now?

Reply:
I updated the code and added tests to cover this use-case.